README.md

```diff
@@ -1,25 +1,147 @@
+`git clone git@github.com:GoogleCloudPlatform/google-cloud-ops-agents-chef.git google_chef_agents`
+from within the cookbooks folder itself.
+
+### Install Cookbook from Source
+- Copy to your Chef server
+  - `knife cookbook upload google_chef_agents`
+- Verify: `knife cookbook show google_chef_agents`
+
+## Usage
+
+| Attribute               | Default       | Description                                                       |
+| ---                     | ---           | ---                                                               |
+| `agent_type`            | Required      | The agent type: `ops-agent`, `monitoring`, `logging`              |
+| `package_state`         | `present`     | Boolean value. Ensure that the agent is `present` or `absent`  |
+| `version`               | `latest`      | The version variable can be used to specify which version of the agent to install. The allowed values are latest, MAJOR_VERSION.*.* and MAJOR_VERSION.MINOR_VERSION.PATCH_VERSION, which are described in detail below. |
+| `main_config`           |               | Optional value for overriding the default configuration           |
+| `additional_config_dir` |               | Optional value for overriding the plugins directory for the `monitoring` or `logging` agents |
+
+### Version
+
+- version=`latest`
+  - This setting makes it easier to keep the agent version up to date, however it does come with a potential risk. When a new major version is released, the policy may install the latest version of the agent from the new major release, which may introduce breaking changes. For production environments, consider using the version=MAJOR_VERSION.*.* setting below for safer agent deployments.
+
+- version=`MAJOR_VERSION.*.*`
+  - When a new major release is out, this setting ensures that only the latest version from the specified major version is installed, which avoids accidentally introducing breaking changes. This is recommended for production environments to ensure safer agent deployments.
+
+- version=`MAJOR_VERSION.MINOR_VERSION.PATCH_VERSION`
+  - This setting is not recommended since it prevents upgrades of new versions of the agent that include bug fixes and other improvements.
+
+### Example:
+Example roles for all below configurations can be found in the examples
+folder. Bootstrapping examples defined below for each configuration. Please
+see the `attributes/default.rb` file or the above section for more information
+on the attributes and their possible values. Their defaults are also
+described.
+
+#### Ops Agent
+Create a role defining the attributes above, and assign it to the node from
+the Chef server or via bootstrap. Bootstrap examples:
+
+- Complete default, this will set version to latest, package_state to present, and agent_type to ops-agent
+  - Linux: knife bootstrap <linux-server-fqdn> -r 'google_chef_agents' -U <username> -P <password> -V
```

Comment:
We do not, as it is the default agent type. See the `attributes/default.rb` file

Reply:
`agent_type` should be a required field to be consistent with Puppet, Terraform and gcloud command.

Reply:
I've switched the behavior to this. I had originally done it with the attribute to set to this because that is Chef standard practice. However, it is something that would almost always be overridden in a role anyway.